back/engines/commercial/insights/app/models/insights/patches/project.rb

```diff
@@ -8,9 +8,16 @@ def self.included(base)
           has_many(
             :insights_data_sources,
             class_name: 'Insights::DataSource',
-            foreign_key: :origin_id,
-            dependent: :destroy
```

Comment:
I would replace `dependent: :destroy` by a comment that explains that data sources will be deleted eventually through views.